src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -267,7 +267,7 @@ internal IVisualStudioHostDocument GetHostDocument(DocumentId documentId)
             }
 
             // first make sure we can edit the document we will be updating (check them out from source control, etc)
-            var changedDocs = projectChanges.SelectMany(pd => pd.GetChangedDocuments(true)).ToList();
+            var changedDocs = projectChanges.SelectMany(pd => pd.GetChangedDocuments(true).Concat(pd.GetChangedAdditionalDocuments())).ToList();
```

Comment:
Note this isn't really the fix: we wouldn't be passing in additional files here, but we still called EnsureEditableDocuments with _just_ the additional file later. But it should have been up-front all along.

Reply:
same here. add like

GetAnyChangedDocuments( ... )